src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs

```diff
@@ -38,6 +38,44 @@ public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> a
         public DocumentBuildContext(string buildOutputFolder, IEnumerable<FileAndType> allSourceFiles, ImmutableArray<string> externalReferencePackages, ImmutableArray<string> xrefMaps, int maxParallelism, string baseFolder, string versionName, ApplyTemplateSettings applyTemplateSetting, string rootTocPath, string versionFolder, ImmutableArray<string> xrefServiceUrls)
             : this(buildOutputFolder, allSourceFiles, externalReferencePackages, xrefMaps, maxParallelism, baseFolder, versionName, applyTemplateSetting, rootTocPath, null, ImmutableArray<string>.Empty, null) { }
 
+        public DocumentBuildContext(DocumentBuildParameters parameters)
+        {
+            BuildOutputFolder = Path.Combine(Directory.GetCurrentDirectory(), parameters.OutputBaseDir);
+            VersionName = parameters.VersionName;
+            ApplyTemplateSettings = parameters.ApplyTemplateSettings;
+            HrefGenerator = parameters.ApplyTemplateSettings?.HrefGenerator;
+            AllSourceFiles = GetAllSourceFiles(parameters.Files.EnumerateFiles());
+            ExternalReferencePackages = parameters.ExternalReferencePackages;
+            _xrefMapUrls = parameters.XRefMaps;
+            _xrefServiceUrls = parameters.XRefServiceUrls;
+            GroupInfo = parameters.GroupInfo;
+            MaxParallelism = parameters.MaxParallelism;
+            MaxHttpParallelism = parameters.MaxHttpParallelism;
+
+            if (parameters.XRefMaps.Length > 0)
+            {
+                _reader = new XRefCollection(
+                    from u in parameters.XRefMaps
+                    select new Uri(u, UriKind.RelativeOrAbsolute)).GetReaderAsync(parameters.Files.DefaultBaseDir);
+            }
+            RootTocPath = parameters.RootTocPath;
+
+            if (!string.IsNullOrEmpty(parameters.VersionDir) && Path.IsPathRooted(parameters.VersionDir))
```

Comment:
>VersionDir [](start = 49, length = 10)

It's deprecated. Can use GroupInfo.Destination

Reply:
same as old constructor.